modules: Facilitate the use of nrfx drivers

anangl opened the pull request:
This set of commits is intended to facilitate the use of nrfx drivers, and hence nRF peripherals, in Zephyr applications, especially in cases when there are no corresponding Zephyr driver shims.

---

Requires https://github.com/zephyrproject-rtos/hal_nordic/pull/2 to go in first.

anangl committed:
modules: Kconfig: Unify the way of enabling instances in nrfx drivers

Use a consistent way of enabling particular instances of peripherals
in nrfx drivers through options defined in modules/Kconfig.nordic,
to make the usage of nrfx drivers in Zephyr, especially the ones for
which there are no Zephyr driver shims (yet), easier.

Jira: NCSDK-2744

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
modules: Kconfig: Make options enabling nrfx drivers easier to access

The options are given prompts, so that they become accessible from
application configuration files and visible in menuconfig, and they
are complemented with proper dependencies, so that only the relevant
ones for the selected SoC are made available.

Jira: NCSDK-2744

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
modules: Kconfig: Add missing options enabling nrfx drivers

Quite a few nrfx drivers are missing corresponding Kconfig options
that would enable their use in Zephyr applications.
Add such options, to facilitate the use of nRF peripherals for which
there are no Zephyr driver shims or even suitable APIs.

Jira: NCSDK-2744

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

ulfalizer commented on modules/Kconfig.nordic:
```diff
@@ -10,121 +10,363 @@ config HAS_NORDIC_DRIVERS
 
-config NRFX_PRS
-	bool
+if HAS_NRFX
```
Can put a `depends on` directly on the menu instead, like

```
menu "nrfx drivers"
	depends on HAS_NRFX

...
```

anangl replied:
Good to know, I'll update it.

ulfalizer commented on modules/Kconfig.nordic:
```diff
@@ -10,121 +10,363 @@ config HAS_NORDIC_DRIVERS
 config NRFX_RTC
-	bool
+	bool "Enable RTC driver"
+	depends on (HAS_HW_NRF_RTC0 || HAS_HW_NRF_RTC1 || HAS_HW_NRF_RTC2)
```
Works without the parens too btw:

```
	depends on HAS_HW_NRF_RTC0 || HAS_HW_NRF_RTC1 || HAS_HW_NRF_RTC2
```

anangl replied:
Ok, I'll remove them.

anangl committed:
modules: hal: nordic: Update the module revision

Update the module revision so that the following commits become
effective:

* nrfx_config: Use common mappings of Kconfig options to nrfx macros
* nrfx_config: Unify the way of enabling instances in nrfx drivers
* nrfx_config: Add missing mappings of Kconfig option symbols
* CMakeLists: Add missing inclusions of driver implementation files

The above changes require a minor correction in the nrf91/soc.h file,
so it is also modified in this commit.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl commented:
In the hal_nordic module I added a protection against including particular nrfx_config_*.h files directly, not through the common nrfx_config.h. I also moved the change done to the west manifest to a separate commit, for clarity.

anangl commented:
Updated _west.yml_ with the proper revision of the hal_nordic module as https://github.com/zephyrproject-rtos/hal_nordic/pull/2 has been merged.

kl-cruz reviewed (approved):
I see no problems here. Approved.
